terraform/slurm_cluster/modules/slurm_nodeset_tpu/main.tf

```diff
@@ -90,17 +87,10 @@ data "google_compute_default_service_account" "this" {
 }
 
 data "google_compute_subnetwork" "nodeset_subnetwork" {
-  count = local.snetwork_valid ? 1 : 0
+  name = var.subnetwork
 
-  project = var.project_id
-  region  = local.region
-  name    = var.subnetwork
-  self_link = (
```

Comment:
Since `region` is removed and no `self_link` is provided, it will default to the region set in the provider.
We should either:
* use `self_link` instead of `name` - discussed offline, will be done later;
* keep specifying `region`

Reply:
Added back region.